
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://ltalignani.github.io/mkdocs-bash-scripts-collection/Vcf/variants-invariants/">
      
      
        <link rel="prev" href="../homopolymer/">
      
      
        <link rel="next" href="../genotype-quality/">
      
      
      <link rel="icon" href="../../assets/favicon.ico">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.15">
    
    
      
        <title>Filtering of Variants & Invariants - Bash Script Collection</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.342714a4.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../css/overrides.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="deep-purple">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#variant-and-invariant-site-filtering-for-population-genomics" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Bash Script Collection" class="md-header__button md-logo" aria-label="Bash Script Collection" data-md-component="logo">
      
  <img src="../../assets/logo_LT.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Bash Script Collection
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Filtering of Variants & Invariants
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="deep-purple"  aria-label="Dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5s-1.65.15-2.39.42zM3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29zm.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14zM20.65 7l-1.77 3.79a7.02 7.02 0 0 0-2.38-4.15zm-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29zM12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="blue" data-md-color-accent="deep-orange"  aria-label="Light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/ltalignani/mkdocs-bash-scripts-collection" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    ltalignani/bash-scripts-collection
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../Basic/basic-bash-commands/" class="md-tabs__link">
          
  
  
    
  
  Basic Bash Commands

        </a>
      </li>
    
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../organize-data/" class="md-tabs__link">
        
  
  
    
  
  Organise Your Data

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../Git/" class="md-tabs__link">
        
  
  
    
  
  Start Using Git

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../advanced-git/" class="md-tabs__link">
        
  
  
    
  
  Advanced Git

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../ci-cd/" class="md-tabs__link">
        
  
  
    
  
  CI/CD

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../Fastq/Fastq/" class="md-tabs__link">
          
  
  
    
  
  Fastq Files

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../Bam/Bam/" class="md-tabs__link">
          
  
  
    
  
  Bam Files

        </a>
      </li>
    
  

      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../Vcf/" class="md-tabs__link">
          
  
  
    
  
  VCF Files

        </a>
      </li>
    
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../about/" class="md-tabs__link">
        
  
  
    
  
  About

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Bash Script Collection" class="md-nav__button md-logo" aria-label="Bash Script Collection" data-md-component="logo">
      
  <img src="../../assets/logo_LT.png" alt="logo">

    </a>
    Bash Script Collection
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/ltalignani/mkdocs-bash-scripts-collection" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    ltalignani/bash-scripts-collection
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Basic Bash Commands
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Basic Bash Commands
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Basic/basic-bash-commands/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Basic/file-management/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    File Management
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Basic/text-processing/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Text Processing
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Basic/system-information/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    System Information
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Basic/networking/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Networking
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Basic/process-management/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Process Management
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../organize-data/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Organise Your Data
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Git/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Start Using Git
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../advanced-git/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Advanced Git
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../ci-cd/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CI/CD
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Fastq Files
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Fastq Files
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Fastq/Fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fastq Format
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Fastq/rename-fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename Fastq Files
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Fastq/FastQC/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    FastQC
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Fastq/Fastq-screen/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fastq-screen
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Fastq/Trimming/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Trimming
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Fastq/BWA/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mapping
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_8" >
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Bam Files
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            Bam Files
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Bam/Bam/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Bam Format
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Bam/sort-bam/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Sorting Bam Files
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Bam/add-or-replace-read-groups/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Add or Replace Read Groups
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Bam/mark-duplicates/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Marking Duplicates
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Bam/merge-and-mark-duplicates/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Merge and Mark Duplicates
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Bam/indel-realignment/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Indel Realignment
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Bam/polishing/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Polishing Bam Files
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Bam/bam-qc/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Bam Quality Control
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Bam/variant-calling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Variant Calling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Bam/genomicsdbimport/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GenomicsDBImport and GenotypeGVCFs
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_9" checked>
        
          
          <label class="md-nav__link" for="__nav_9" id="__nav_9_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    VCF Files
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_9_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_9">
            <span class="md-nav__icon md-icon"></span>
            VCF Files
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Vcf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VCF Format
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../accessibility/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Filtering by Accessibility
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../homopolymer/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Filtering by Homopolymer
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Filtering of Variants & Invariants
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Filtering of Variants & Invariants
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    <span class="md-ellipsis">
      Overview
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rationale-for-variant-invariant-separation" class="md-nav__link">
    <span class="md-ellipsis">
      Rationale for Variant-Invariant Separation
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#implementation-workflow" class="md-nav__link">
    <span class="md-ellipsis">
      Implementation Workflow
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#population-genomics-applications" class="md-nav__link">
    <span class="md-ellipsis">
      Population Genomics Applications
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#best-practices-and-recommendations" class="md-nav__link">
    <span class="md-ellipsis">
      Best Practices and Recommendations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#limitations-and-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      Limitations and Considerations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#future-developments" class="md-nav__link">
    <span class="md-ellipsis">
      Future Developments
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#conclusion" class="md-nav__link">
    <span class="md-ellipsis">
      Conclusion
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../genotype-quality/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Filtering by Genotype Quality
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../concatenate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VCF concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../gt-indv/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Filtering by GT and INDV
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../about/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    About
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="variant-and-invariant-site-filtering-for-population-genomics">Variant and Invariant Site Filtering for Population Genomics<a class="headerlink" href="#variant-and-invariant-site-filtering-for-population-genomics" title="Permanent link">&para;</a></h1>
<h2 class="no-toc">Table of Content</h2>

<div class="toc">
<ul>
<li><a href="#variant-and-invariant-site-filtering-for-population-genomics">Variant and Invariant Site Filtering for Population Genomics</a><ul>
<li><a href="#overview">Overview</a></li>
<li><a href="#rationale-for-variant-invariant-separation">Rationale for Variant-Invariant Separation</a></li>
<li><a href="#implementation-workflow">Implementation Workflow</a></li>
<li><a href="#population-genomics-applications">Population Genomics Applications</a></li>
<li><a href="#best-practices-and-recommendations">Best Practices and Recommendations</a></li>
<li><a href="#limitations-and-considerations">Limitations and Considerations</a></li>
<li><a href="#future-developments">Future Developments</a></li>
<li><a href="#conclusion">Conclusion</a></li>
</ul>
</li>
</ul>
</div>
<h2 id="overview">Overview<a class="headerlink" href="#overview" title="Permanent link">&para;</a></h2>
<p>In population genomics analyses, the accurate characterization of genetic diversity requires comprehensive assessment of both polymorphic (variant) and monomorphic (invariant) sites across the genome. Traditional variant calling pipelines focus primarily on identifying and filtering variable positions, often discarding invariant sites that are crucial for calculating accurate population genetic parameters such as nucleotide diversity (π), Tajima's D, and demographic inference statistics.</p>
<h2 id="rationale-for-variant-invariant-separation">Rationale for Variant-Invariant Separation<a class="headerlink" href="#rationale-for-variant-invariant-separation" title="Permanent link">&para;</a></h2>
<h3 id="the-importance-of-invariant-sites">The Importance of Invariant Sites<a class="headerlink" href="#the-importance-of-invariant-sites" title="Permanent link">&para;</a></h3>
<p>Invariant sites represent genomic positions where all sequenced individuals share the same allele. These sites are essential for:</p>
<p><strong>Population Genetic Statistics:</strong><br />
- <strong>Nucleotide diversity calculations</strong>: Accurate π estimates require knowledge of both polymorphic and monomorphic sites<br />
- <strong>Site frequency spectrum</strong>: Complete allele frequency distributions need invariant site counts<br />
- <strong>Demographic inference</strong>: Coalescent-based methods require total sequence length information<br />
- <strong>Selection analysis</strong>: Identifying regions under purifying selection through reduced diversity  </p>
<p><strong>Comparative Genomics:</strong><br />
- <strong>Divergence estimates</strong>: Calculating substitution rates between populations or species<br />
- <strong>Conservation analysis</strong>: Identifying functionally constrained genomic regions<br />
- <strong>Mutation rate estimation</strong>: Determining background mutation rates across different genomic contexts  </p>
<h3 id="differential-filtering-requirements">Differential Filtering Requirements<a class="headerlink" href="#differential-filtering-requirements" title="Permanent link">&para;</a></h3>
<p>Variants and invariants require distinct filtering strategies due to their fundamental differences:  </p>
<p><strong>Variant Sites:</strong><br />
- Higher susceptibility to technical artifacts<br />
- Complex allelic configurations requiring sophisticated quality metrics<br />
- Strand bias and mapping quality considerations<br />
- Allele balance and Hardy-Weinberg equilibrium assessment  </p>
<p><strong>Invariant Sites:</strong><br />
- Simpler quality assessment based primarily on coverage depth<br />
- Lower risk of systematic bias<br />
- Focused on sequencing completeness rather than allelic accuracy<br />
- Reduced computational complexity for quality control  </p>
<h2 id="implementation-workflow">Implementation Workflow<a class="headerlink" href="#implementation-workflow" title="Permanent link">&para;</a></h2>
<h3 id="step-1-variant-and-invariant-separation">Step 1: Variant and Invariant Separation<a class="headerlink" href="#step-1-variant-and-invariant-separation" title="Permanent link">&para;</a></h3>
<p>The initial step involves separating variant and invariant sites into distinct VCF files for independent processing:</p>
<div class="language-bash highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-0-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-0-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-0-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-0-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-0-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-0-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-0-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-0-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-0-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-0-10">10</a></span>
<span class="normal"><a href="#__codelineno-0-11">11</a></span>
<span class="normal"><a href="#__codelineno-0-12">12</a></span>
<span class="normal"><a href="#__codelineno-0-13">13</a></span>
<span class="normal"><a href="#__codelineno-0-14">14</a></span>
<span class="normal"><a href="#__codelineno-0-15">15</a></span>
<span class="normal"><a href="#__codelineno-0-16">16</a></span>
<span class="normal"><a href="#__codelineno-0-17">17</a></span>
<span class="normal"><a href="#__codelineno-0-18">18</a></span>
<span class="normal"><a href="#__codelineno-0-19">19</a></span>
<span class="normal"><a href="#__codelineno-0-20">20</a></span>
<span class="normal"><a href="#__codelineno-0-21">21</a></span>
<span class="normal"><a href="#__codelineno-0-22">22</a></span>
<span class="normal"><a href="#__codelineno-0-23">23</a></span>
<span class="normal"><a href="#__codelineno-0-24">24</a></span>
<span class="normal"><a href="#__codelineno-0-25">25</a></span>
<span class="normal"><a href="#__codelineno-0-26">26</a></span>
<span class="normal"><a href="#__codelineno-0-27">27</a></span>
<span class="normal"><a href="#__codelineno-0-28">28</a></span>
<span class="normal"><a href="#__codelineno-0-29">29</a></span>
<span class="normal"><a href="#__codelineno-0-30">30</a></span>
<span class="normal"><a href="#__codelineno-0-31">31</a></span>
<span class="normal"><a href="#__codelineno-0-32">32</a></span>
<span class="normal"><a href="#__codelineno-0-33">33</a></span>
<span class="normal"><a href="#__codelineno-0-34">34</a></span>
<span class="normal"><a href="#__codelineno-0-35">35</a></span>
<span class="normal"><a href="#__codelineno-0-36">36</a></span>
<span class="normal"><a href="#__codelineno-0-37">37</a></span>
<span class="normal"><a href="#__codelineno-0-38">38</a></span>
<span class="normal"><a href="#__codelineno-0-39">39</a></span>
<span class="normal"><a href="#__codelineno-0-40">40</a></span>
<span class="normal"><a href="#__codelineno-0-41">41</a></span>
<span class="normal"><a href="#__codelineno-0-42">42</a></span>
<span class="normal"><a href="#__codelineno-0-43">43</a></span>
<span class="normal"><a href="#__codelineno-0-44">44</a></span>
<span class="normal"><a href="#__codelineno-0-45">45</a></span>
<span class="normal"><a href="#__codelineno-0-46">46</a></span>
<span class="normal"><a href="#__codelineno-0-47">47</a></span>
<span class="normal"><a href="#__codelineno-0-48">48</a></span>
<span class="normal"><a href="#__codelineno-0-49">49</a></span>
<span class="normal"><a href="#__codelineno-0-50">50</a></span>
<span class="normal"><a href="#__codelineno-0-51">51</a></span>
<span class="normal"><a href="#__codelineno-0-52">52</a></span>
<span class="normal"><a href="#__codelineno-0-53">53</a></span></pre></div></td><td class="code"><div><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2"></a><span class="c1">###################configuration slurm##############################</span>
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3"></a><span class="c1">#SBATCH -A invalbo</span>
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4"></a><span class="c1">#SBATCH --time=12:00:00</span>
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5"></a><span class="c1">#SBATCH --job-name=var_invar</span>
</span><span id="__span-0-6"><a id="__codelineno-0-6" name="__codelineno-0-6"></a><span class="c1">#SBATCH -p fast</span>
</span><span id="__span-0-7"><a id="__codelineno-0-7" name="__codelineno-0-7"></a><span class="c1">#SBATCH --nodes=1</span>
</span><span id="__span-0-8"><a id="__codelineno-0-8" name="__codelineno-0-8"></a><span class="c1">#SBATCH --ntasks=1</span>
</span><span id="__span-0-9"><a id="__codelineno-0-9" name="__codelineno-0-9"></a><span class="c1">#SBATCH --cpus-per-task=8</span>
</span><span id="__span-0-10"><a id="__codelineno-0-10" name="__codelineno-0-10"></a><span class="c1">#SBATCH --mem=16G</span>
</span><span id="__span-0-11"><a id="__codelineno-0-11" name="__codelineno-0-11"></a><span class="c1">#SBATCH --array=1-4</span>
</span><span id="__span-0-12"><a id="__codelineno-0-12" name="__codelineno-0-12"></a><span class="c1">#SBATCH --output=Cluster_logs/%x-%j-%N.out</span>
</span><span id="__span-0-13"><a id="__codelineno-0-13" name="__codelineno-0-13"></a><span class="c1">#SBATCH --error=Cluster_logs/%x-%j-%N.err</span>
</span><span id="__span-0-14"><a id="__codelineno-0-14" name="__codelineno-0-14"></a><span class="c1">#SBATCH --mail-user=loic.talignani@ird.fr</span>
</span><span id="__span-0-15"><a id="__codelineno-0-15" name="__codelineno-0-15"></a><span class="c1">#SBATCH --mail-type=FAIL</span>
</span><span id="__span-0-16"><a id="__codelineno-0-16" name="__codelineno-0-16"></a><span class="c1">####################################################################</span>
</span><span id="__span-0-17"><a id="__codelineno-0-17" name="__codelineno-0-17"></a>
</span><span id="__span-0-18"><a id="__codelineno-0-18" name="__codelineno-0-18"></a><span class="nv">start_time</span><span class="o">=</span><span class="k">$(</span>date<span class="w"> </span>+%s<span class="k">)</span>
</span><span id="__span-0-19"><a id="__codelineno-0-19" name="__codelineno-0-19"></a>
</span><span id="__span-0-20"><a id="__codelineno-0-20" name="__codelineno-0-20"></a><span class="c1"># Récupération du nom du fichier BAM à partir du fichier de liste</span>
</span><span id="__span-0-21"><a id="__codelineno-0-21" name="__codelineno-0-21"></a><span class="nv">CHROM</span><span class="o">=</span><span class="k">$(</span>sed<span class="w"> </span>-n<span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span><span class="s2">p&quot;</span><span class="w"> </span>info_files/chrom.list<span class="k">)</span>
</span><span id="__span-0-22"><a id="__codelineno-0-22" name="__codelineno-0-22"></a>
</span><span id="__span-0-23"><a id="__codelineno-0-23" name="__codelineno-0-23"></a><span class="c1"># Répertoire de sortie pour les fichiers</span>
</span><span id="__span-0-24"><a id="__codelineno-0-24" name="__codelineno-0-24"></a><span class="nv">INPUT_DIR</span><span class="o">=</span><span class="s2">&quot;vcfs_flt_HRun&quot;</span>
</span><span id="__span-0-25"><a id="__codelineno-0-25" name="__codelineno-0-25"></a>
</span><span id="__span-0-26"><a id="__codelineno-0-26" name="__codelineno-0-26"></a><span class="nv">OUTPUT_DIR</span><span class="o">=</span><span class="s2">&quot;vcfs_flt_HRun_slct_vrt&quot;</span>
</span><span id="__span-0-27"><a id="__codelineno-0-27" name="__codelineno-0-27"></a>mkdir<span class="w"> </span>-p<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>
</span><span id="__span-0-28"><a id="__codelineno-0-28" name="__codelineno-0-28"></a>
</span><span id="__span-0-29"><a id="__codelineno-0-29" name="__codelineno-0-29"></a><span class="nv">LOG_DIR</span><span class="o">=</span><span class="s2">&quot;logs&quot;</span>
</span><span id="__span-0-30"><a id="__codelineno-0-30" name="__codelineno-0-30"></a>mkdir<span class="w"> </span>-p<span class="w"> </span><span class="si">${</span><span class="nv">LOG_DIR</span><span class="si">}</span>
</span><span id="__span-0-31"><a id="__codelineno-0-31" name="__codelineno-0-31"></a>
</span><span id="__span-0-32"><a id="__codelineno-0-32" name="__codelineno-0-32"></a><span class="c1"># Load module</span>
</span><span id="__span-0-33"><a id="__codelineno-0-33" name="__codelineno-0-33"></a>module<span class="w"> </span>load<span class="w"> </span>gatk4/4.2.6.1
</span><span id="__span-0-34"><a id="__codelineno-0-34" name="__codelineno-0-34"></a>
</span><span id="__span-0-35"><a id="__codelineno-0-35" name="__codelineno-0-35"></a>gatk<span class="w"> </span>IndexFeatureFile<span class="w"> </span>-I<span class="w"> </span><span class="si">${</span><span class="nv">INPUT_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.vcf.gz
</span><span id="__span-0-36"><a id="__codelineno-0-36" name="__codelineno-0-36"></a>
</span><span id="__span-0-37"><a id="__codelineno-0-37" name="__codelineno-0-37"></a>gatk<span class="w"> </span>SelectVariants<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-38"><a id="__codelineno-0-38" name="__codelineno-0-38"></a><span class="w"> </span>-V<span class="w"> </span><span class="si">${</span><span class="nv">INPUT_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-39"><a id="__codelineno-0-39" name="__codelineno-0-39"></a><span class="w"> </span>-select-type<span class="w"> </span>SNP<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-40"><a id="__codelineno-0-40" name="__codelineno-0-40"></a><span class="w"> </span>-xl-select-type<span class="w"> </span>INDEL<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-41"><a id="__codelineno-0-41" name="__codelineno-0-41"></a><span class="w"> </span>-O<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.VARIANTS.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-42"><a id="__codelineno-0-42" name="__codelineno-0-42"></a><span class="w"> </span><span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="si">${</span><span class="nv">LOG_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.VARIANTS.log
</span><span id="__span-0-43"><a id="__codelineno-0-43" name="__codelineno-0-43"></a>
</span><span id="__span-0-44"><a id="__codelineno-0-44" name="__codelineno-0-44"></a>gatk<span class="w"> </span>SelectVariants<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-45"><a id="__codelineno-0-45" name="__codelineno-0-45"></a><span class="w"> </span>-V<span class="w"> </span><span class="si">${</span><span class="nv">INPUT_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-46"><a id="__codelineno-0-46" name="__codelineno-0-46"></a><span class="w"> </span>-select-type<span class="w"> </span>NO_VARIATION<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-47"><a id="__codelineno-0-47" name="__codelineno-0-47"></a><span class="w"> </span>-xl-select-type<span class="w"> </span>INDEL<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-48"><a id="__codelineno-0-48" name="__codelineno-0-48"></a><span class="w"> </span>-O<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.INVARIANTS.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-49"><a id="__codelineno-0-49" name="__codelineno-0-49"></a><span class="w"> </span><span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="si">${</span><span class="nv">LOG_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.INVARIANTS.log
</span><span id="__span-0-50"><a id="__codelineno-0-50" name="__codelineno-0-50"></a>
</span><span id="__span-0-51"><a id="__codelineno-0-51" name="__codelineno-0-51"></a><span class="nv">end_time</span><span class="o">=</span><span class="k">$(</span>date<span class="w"> </span>+%s<span class="k">)</span><span class="w">  </span><span class="c1"># Capture l&#39;heure de fin</span>
</span><span id="__span-0-52"><a id="__codelineno-0-52" name="__codelineno-0-52"></a><span class="nv">duration</span><span class="o">=</span><span class="k">$((</span><span class="nv">end_time</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="nv">start_time</span><span class="k">))</span><span class="w">  </span><span class="c1"># Calcule la durée</span>
</span><span id="__span-0-53"><a id="__codelineno-0-53" name="__codelineno-0-53"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;The script completed successfully in </span><span class="k">$((</span><span class="nv">duration</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="m">60</span><span class="k">))</span><span class="s2"> minutes and </span><span class="k">$((</span><span class="nv">duration</span><span class="w"> </span><span class="o">%</span><span class="w"> </span><span class="m">60</span><span class="k">))</span><span class="s2"> seconds.&quot;</span>
</span></code></pre></div></td></tr></table></div>
<h4 id="technical-details">Technical Details<a class="headerlink" href="#technical-details" title="Permanent link">&para;</a></h4>
<p><strong>GATK SelectVariants Parameters:</strong><br />
- <code>-select-type SNP</code>: Selects only single nucleotide polymorphisms for variant file<br />
- <code>-select-type NO_VARIATION</code>: Selects invariant sites for invariant file<br />
- <code>-xl-select-type INDEL</code>: Excludes insertion/deletion variants from both outputs<br />
- <code>-V</code>: Input VCF file path<br />
- <code>-O</code>: Output VCF file path  </p>
<p><strong>File Organization:</strong><br />
- Separate logging for variants and invariants facilitates troubleshooting<br />
- Chromosome-specific processing enables parallel execution<br />
- Compressed VCF output reduces storage requirements  </p>
<h3 id="step-2-intermediate-statistics-generation">Step 2: Intermediate Statistics Generation<a class="headerlink" href="#step-2-intermediate-statistics-generation" title="Permanent link">&para;</a></h3>
<p>Quality assessment is performed on both variant and invariant datasets to inform filtering decisions:</p>
<div class="language-bash highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-1-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-1-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-1-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-1-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-1-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-1-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-1-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-1-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-1-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-1-10">10</a></span>
<span class="normal"><a href="#__codelineno-1-11">11</a></span>
<span class="normal"><a href="#__codelineno-1-12">12</a></span>
<span class="normal"><a href="#__codelineno-1-13">13</a></span>
<span class="normal"><a href="#__codelineno-1-14">14</a></span>
<span class="normal"><a href="#__codelineno-1-15">15</a></span>
<span class="normal"><a href="#__codelineno-1-16">16</a></span>
<span class="normal"><a href="#__codelineno-1-17">17</a></span>
<span class="normal"><a href="#__codelineno-1-18">18</a></span>
<span class="normal"><a href="#__codelineno-1-19">19</a></span>
<span class="normal"><a href="#__codelineno-1-20">20</a></span>
<span class="normal"><a href="#__codelineno-1-21">21</a></span>
<span class="normal"><a href="#__codelineno-1-22">22</a></span>
<span class="normal"><a href="#__codelineno-1-23">23</a></span>
<span class="normal"><a href="#__codelineno-1-24">24</a></span>
<span class="normal"><a href="#__codelineno-1-25">25</a></span>
<span class="normal"><a href="#__codelineno-1-26">26</a></span>
<span class="normal"><a href="#__codelineno-1-27">27</a></span>
<span class="normal"><a href="#__codelineno-1-28">28</a></span>
<span class="normal"><a href="#__codelineno-1-29">29</a></span>
<span class="normal"><a href="#__codelineno-1-30">30</a></span>
<span class="normal"><a href="#__codelineno-1-31">31</a></span>
<span class="normal"><a href="#__codelineno-1-32">32</a></span>
<span class="normal"><a href="#__codelineno-1-33">33</a></span>
<span class="normal"><a href="#__codelineno-1-34">34</a></span>
<span class="normal"><a href="#__codelineno-1-35">35</a></span></pre></div></td><td class="code"><div><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2"></a><span class="c1">###################configuration slurm##############################</span>
</span><span id="__span-1-3"><a id="__codelineno-1-3" name="__codelineno-1-3"></a><span class="c1">#SBATCH -A invalbo</span>
</span><span id="__span-1-4"><a id="__codelineno-1-4" name="__codelineno-1-4"></a><span class="c1">#SBATCH --time=01:30:00</span>
</span><span id="__span-1-5"><a id="__codelineno-1-5" name="__codelineno-1-5"></a><span class="c1">#SBATCH --job-name=stats</span>
</span><span id="__span-1-6"><a id="__codelineno-1-6" name="__codelineno-1-6"></a><span class="c1">#SBATCH -p fast</span>
</span><span id="__span-1-7"><a id="__codelineno-1-7" name="__codelineno-1-7"></a><span class="c1">#SBATCH --nodes=1</span>
</span><span id="__span-1-8"><a id="__codelineno-1-8" name="__codelineno-1-8"></a><span class="c1">#SBATCH --ntasks=1</span>
</span><span id="__span-1-9"><a id="__codelineno-1-9" name="__codelineno-1-9"></a><span class="c1">#SBATCH --cpus-per-task=1</span>
</span><span id="__span-1-10"><a id="__codelineno-1-10" name="__codelineno-1-10"></a><span class="c1">#SBATCH --mem=8G</span>
</span><span id="__span-1-11"><a id="__codelineno-1-11" name="__codelineno-1-11"></a><span class="c1">#SBATCH --array=1-5</span>
</span><span id="__span-1-12"><a id="__codelineno-1-12" name="__codelineno-1-12"></a><span class="c1">#SBATCH --output=Cluster_logs/%x-%j-%N.out</span>
</span><span id="__span-1-13"><a id="__codelineno-1-13" name="__codelineno-1-13"></a><span class="c1">#SBATCH --error=Cluster_logs/%x-%j-%N.err</span>
</span><span id="__span-1-14"><a id="__codelineno-1-14" name="__codelineno-1-14"></a><span class="c1">#SBATCH --mail-user=loic.talignani@ird.fr</span>
</span><span id="__span-1-15"><a id="__codelineno-1-15" name="__codelineno-1-15"></a><span class="c1">#SBATCH --mail-type=FAIL</span>
</span><span id="__span-1-16"><a id="__codelineno-1-16" name="__codelineno-1-16"></a><span class="c1">####################################################################</span>
</span><span id="__span-1-17"><a id="__codelineno-1-17" name="__codelineno-1-17"></a>
</span><span id="__span-1-18"><a id="__codelineno-1-18" name="__codelineno-1-18"></a><span class="nv">start_time</span><span class="o">=</span><span class="k">$(</span>date<span class="w"> </span>+%s<span class="k">)</span>
</span><span id="__span-1-19"><a id="__codelineno-1-19" name="__codelineno-1-19"></a>
</span><span id="__span-1-20"><a id="__codelineno-1-20" name="__codelineno-1-20"></a><span class="c1"># Récupération du nom du fichier BAM à partir du fichier de liste</span>
</span><span id="__span-1-21"><a id="__codelineno-1-21" name="__codelineno-1-21"></a><span class="nv">CHROM</span><span class="o">=</span><span class="k">$(</span>sed<span class="w"> </span>-n<span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span><span class="s2">p&quot;</span><span class="w"> </span>info_files/chrom.list<span class="k">)</span>
</span><span id="__span-1-22"><a id="__codelineno-1-22" name="__codelineno-1-22"></a>
</span><span id="__span-1-23"><a id="__codelineno-1-23" name="__codelineno-1-23"></a><span class="c1"># Répertoire de sortie pour les fichiers</span>
</span><span id="__span-1-24"><a id="__codelineno-1-24" name="__codelineno-1-24"></a><span class="nv">OUTPUT_DIR</span><span class="o">=</span><span class="s2">&quot;vcfs_flt_HRun_slct_vrt&quot;</span>
</span><span id="__span-1-25"><a id="__codelineno-1-25" name="__codelineno-1-25"></a>
</span><span id="__span-1-26"><a id="__codelineno-1-26" name="__codelineno-1-26"></a><span class="c1"># Load module</span>
</span><span id="__span-1-27"><a id="__codelineno-1-27" name="__codelineno-1-27"></a>module<span class="w"> </span>load<span class="w"> </span>bcftools/1.15.1
</span><span id="__span-1-28"><a id="__codelineno-1-28" name="__codelineno-1-28"></a>
</span><span id="__span-1-29"><a id="__codelineno-1-29" name="__codelineno-1-29"></a>bcftools<span class="w"> </span>stats<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.INVARIANTS.vcf.gz<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.INVARIANTS.stats
</span><span id="__span-1-30"><a id="__codelineno-1-30" name="__codelineno-1-30"></a>
</span><span id="__span-1-31"><a id="__codelineno-1-31" name="__codelineno-1-31"></a>bcftools<span class="w"> </span>stats<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.VARIANTS.vcf.gz<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.VARIANTS.stats
</span><span id="__span-1-32"><a id="__codelineno-1-32" name="__codelineno-1-32"></a>
</span><span id="__span-1-33"><a id="__codelineno-1-33" name="__codelineno-1-33"></a><span class="nv">end_time</span><span class="o">=</span><span class="k">$(</span>date<span class="w"> </span>+%s<span class="k">)</span><span class="w">  </span><span class="c1"># Capture l&#39;heure de fin</span>
</span><span id="__span-1-34"><a id="__codelineno-1-34" name="__codelineno-1-34"></a><span class="nv">duration</span><span class="o">=</span><span class="k">$((</span><span class="nv">end_time</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="nv">start_time</span><span class="k">))</span><span class="w">  </span><span class="c1"># Calcule la durée</span>
</span><span id="__span-1-35"><a id="__codelineno-1-35" name="__codelineno-1-35"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;The script completed successfully in </span><span class="k">$((</span><span class="nv">duration</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="m">60</span><span class="k">))</span><span class="s2"> minutes and </span><span class="k">$((</span><span class="nv">duration</span><span class="w"> </span><span class="o">%</span><span class="w"> </span><span class="m">60</span><span class="k">))</span><span class="s2"> seconds.&quot;</span>
</span></code></pre></div></td></tr></table></div>
<h4 id="statistical-analysis">Statistical Analysis<a class="headerlink" href="#statistical-analysis" title="Permanent link">&para;</a></h4>
<p><strong>BCFtools stats output includes:</strong><br />
- <strong>Site counts</strong>: Total number of variants and invariants<br />
- <strong>Quality distributions</strong>: QUAL score histograms for threshold selection<br />
- <strong>Depth statistics</strong>: Coverage depth distributions across sites<br />
- <strong>Allele frequency spectra</strong>: Distribution of minor allele frequencies (variants only)<br />
- <strong>Transition/transversion ratios</strong>: Mutation pattern assessment (variants only)  </p>
<h3 id="step-3-differential-filtering-and-flagging">Step 3: Differential Filtering and Flagging<a class="headerlink" href="#step-3-differential-filtering-and-flagging" title="Permanent link">&para;</a></h3>
<p>Variants and invariants are subjected to distinct filtering criteria reflecting their different quality requirements.</p>
<p>Before applying any filtering criteria, it is interesting to examine the distribution of DP (depth of coverage) values in each VCF file. This data-driven approach helps determine appropriate thresholds for filtering both variant and invariant sites, ensuring that quality filters are adapted to the characteristics of the dataset.</p>
<div class="language-python highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-2-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-2-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-2-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-2-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-2-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-2-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-2-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-2-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-2-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-2-10">10</a></span>
<span class="normal"><a href="#__codelineno-2-11">11</a></span>
<span class="normal"><a href="#__codelineno-2-12">12</a></span>
<span class="normal"><a href="#__codelineno-2-13">13</a></span>
<span class="normal"><a href="#__codelineno-2-14">14</a></span>
<span class="normal"><a href="#__codelineno-2-15">15</a></span>
<span class="normal"><a href="#__codelineno-2-16">16</a></span>
<span class="normal"><a href="#__codelineno-2-17">17</a></span>
<span class="normal"><a href="#__codelineno-2-18">18</a></span>
<span class="normal"><a href="#__codelineno-2-19">19</a></span>
<span class="normal"><a href="#__codelineno-2-20">20</a></span>
<span class="normal"><a href="#__codelineno-2-21">21</a></span>
<span class="normal"><a href="#__codelineno-2-22">22</a></span>
<span class="normal"><a href="#__codelineno-2-23">23</a></span>
<span class="normal"><a href="#__codelineno-2-24">24</a></span>
<span class="normal"><a href="#__codelineno-2-25">25</a></span>
<span class="normal"><a href="#__codelineno-2-26">26</a></span>
<span class="normal"><a href="#__codelineno-2-27">27</a></span>
<span class="normal"><a href="#__codelineno-2-28">28</a></span>
<span class="normal"><a href="#__codelineno-2-29">29</a></span>
<span class="normal"><a href="#__codelineno-2-30">30</a></span>
<span class="normal"><a href="#__codelineno-2-31">31</a></span>
<span class="normal"><a href="#__codelineno-2-32">32</a></span>
<span class="normal"><a href="#__codelineno-2-33">33</a></span>
<span class="normal"><a href="#__codelineno-2-34">34</a></span>
<span class="normal"><a href="#__codelineno-2-35">35</a></span>
<span class="normal"><a href="#__codelineno-2-36">36</a></span>
<span class="normal"><a href="#__codelineno-2-37">37</a></span>
<span class="normal"><a href="#__codelineno-2-38">38</a></span>
<span class="normal"><a href="#__codelineno-2-39">39</a></span>
<span class="normal"><a href="#__codelineno-2-40">40</a></span>
<span class="normal"><a href="#__codelineno-2-41">41</a></span></pre></div></td><td class="code"><div><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1"></a><span class="n">pip</span> <span class="n">install</span> <span class="n">pysam</span> <span class="n">numpy</span>
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2"></a><span class="kn">import</span><span class="w"> </span><span class="nn">pysam</span>
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3"></a><span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
</span><span id="__span-2-4"><a id="__codelineno-2-4" name="__codelineno-2-4"></a><span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>
</span><span id="__span-2-5"><a id="__codelineno-2-5" name="__codelineno-2-5"></a>
</span><span id="__span-2-6"><a id="__codelineno-2-6" name="__codelineno-2-6"></a><span class="c1"># Load VCF file</span>
</span><span id="__span-2-7"><a id="__codelineno-2-7" name="__codelineno-2-7"></a><span class="n">vcf_file</span> <span class="o">=</span> <span class="s2">&quot;combined.chr2L.acc.HRun.VARIANTS.vcf.gz&quot;</span>
</span><span id="__span-2-8"><a id="__codelineno-2-8" name="__codelineno-2-8"></a>
</span><span id="__span-2-9"><a id="__codelineno-2-9" name="__codelineno-2-9"></a><span class="c1"># Define intervals</span>
</span><span id="__span-2-10"><a id="__codelineno-2-10" name="__codelineno-2-10"></a><span class="n">bins</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1551</span><span class="p">,</span> <span class="mi">50</span><span class="p">))</span>
</span><span id="__span-2-11"><a id="__codelineno-2-11" name="__codelineno-2-11"></a><span class="n">interval_labels</span> <span class="o">=</span> <span class="p">[</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">bins</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="si">}</span><span class="s2">-</span><span class="si">{</span><span class="n">bins</span><span class="p">[</span><span class="n">i</span><span class="o">+</span><span class="mi">1</span><span class="p">]</span><span class="o">-</span><span class="mi">1</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">bins</span><span class="p">)</span><span class="o">-</span><span class="mi">1</span><span class="p">)]</span>
</span><span id="__span-2-12"><a id="__codelineno-2-12" name="__codelineno-2-12"></a>
</span><span id="__span-2-13"><a id="__codelineno-2-13" name="__codelineno-2-13"></a><span class="c1"># Initialize dict to count frequencies</span>
</span><span id="__span-2-14"><a id="__codelineno-2-14" name="__codelineno-2-14"></a><span class="n">dp_counts</span> <span class="o">=</span> <span class="p">{</span><span class="n">label</span><span class="p">:</span> <span class="mi">0</span> <span class="k">for</span> <span class="n">label</span> <span class="ow">in</span> <span class="n">interval_labels</span><span class="p">}</span>
</span><span id="__span-2-15"><a id="__codelineno-2-15" name="__codelineno-2-15"></a>
</span><span id="__span-2-16"><a id="__codelineno-2-16" name="__codelineno-2-16"></a><span class="c1"># Read VCF file</span>
</span><span id="__span-2-17"><a id="__codelineno-2-17" name="__codelineno-2-17"></a><span class="k">with</span> <span class="n">pysam</span><span class="o">.</span><span class="n">VariantFile</span><span class="p">(</span><span class="n">vcf_file</span><span class="p">,</span> <span class="s2">&quot;r&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">vcf</span><span class="p">:</span>
</span><span id="__span-2-18"><a id="__codelineno-2-18" name="__codelineno-2-18"></a>    <span class="k">for</span> <span class="n">record</span> <span class="ow">in</span> <span class="n">vcf</span><span class="p">:</span>
</span><span id="__span-2-19"><a id="__codelineno-2-19" name="__codelineno-2-19"></a>        <span class="n">dp</span> <span class="o">=</span> <span class="n">record</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;DP&quot;</span><span class="p">)</span>
</span><span id="__span-2-20"><a id="__codelineno-2-20" name="__codelineno-2-20"></a>        <span class="k">if</span> <span class="n">dp</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="__span-2-21"><a id="__codelineno-2-21" name="__codelineno-2-21"></a>            <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="p">(</span><span class="n">low</span><span class="p">,</span> <span class="n">high</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">bins</span><span class="p">[:</span><span class="o">-</span><span class="mi">1</span><span class="p">],</span> <span class="n">bins</span><span class="p">[</span><span class="mi">1</span><span class="p">:])):</span>
</span><span id="__span-2-22"><a id="__codelineno-2-22" name="__codelineno-2-22"></a>                <span class="k">if</span> <span class="n">low</span> <span class="o">&lt;=</span> <span class="n">dp</span> <span class="o">&lt;</span> <span class="n">high</span><span class="p">:</span>
</span><span id="__span-2-23"><a id="__codelineno-2-23" name="__codelineno-2-23"></a>                    <span class="n">dp_counts</span><span class="p">[</span><span class="n">interval_labels</span><span class="p">[</span><span class="n">i</span><span class="p">]]</span> <span class="o">+=</span> <span class="mi">1</span>
</span><span id="__span-2-24"><a id="__codelineno-2-24" name="__codelineno-2-24"></a>                    <span class="k">break</span>
</span><span id="__span-2-25"><a id="__codelineno-2-25" name="__codelineno-2-25"></a>
</span><span id="__span-2-26"><a id="__codelineno-2-26" name="__codelineno-2-26"></a><span class="c1"># Prepare data for graph</span>
</span><span id="__span-2-27"><a id="__codelineno-2-27" name="__codelineno-2-27"></a><span class="n">intervals</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">dp_counts</span><span class="o">.</span><span class="n">keys</span><span class="p">())</span>
</span><span id="__span-2-28"><a id="__codelineno-2-28" name="__codelineno-2-28"></a><span class="n">frequencies</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">dp_counts</span><span class="o">.</span><span class="n">values</span><span class="p">())</span>
</span><span id="__span-2-29"><a id="__codelineno-2-29" name="__codelineno-2-29"></a>
</span><span id="__span-2-30"><a id="__codelineno-2-30" name="__codelineno-2-30"></a><span class="c1"># Draw graph</span>
</span><span id="__span-2-31"><a id="__codelineno-2-31" name="__codelineno-2-31"></a><span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">6</span><span class="p">))</span>
</span><span id="__span-2-32"><a id="__codelineno-2-32" name="__codelineno-2-32"></a><span class="n">plt</span><span class="o">.</span><span class="n">bar</span><span class="p">(</span><span class="n">intervals</span><span class="p">,</span> <span class="n">frequencies</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s2">&quot;skyblue&quot;</span><span class="p">,</span> <span class="n">edgecolor</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">)</span>
</span><span id="__span-2-33"><a id="__codelineno-2-33" name="__codelineno-2-33"></a><span class="n">plt</span><span class="o">.</span><span class="n">xticks</span><span class="p">(</span><span class="n">rotation</span><span class="o">=</span><span class="mi">45</span><span class="p">)</span>
</span><span id="__span-2-34"><a id="__codelineno-2-34" name="__codelineno-2-34"></a><span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;Intervalle de DP&quot;</span><span class="p">)</span>
</span><span id="__span-2-35"><a id="__codelineno-2-35" name="__codelineno-2-35"></a><span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;Fréquence&quot;</span><span class="p">)</span>
</span><span id="__span-2-36"><a id="__codelineno-2-36" name="__codelineno-2-36"></a><span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s2">&quot;Distribution des valeurs de DP dans le champ INFO&quot;</span><span class="p">)</span>
</span><span id="__span-2-37"><a id="__codelineno-2-37" name="__codelineno-2-37"></a><span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</span><span id="__span-2-38"><a id="__codelineno-2-38" name="__codelineno-2-38"></a>
</span><span id="__span-2-39"><a id="__codelineno-2-39" name="__codelineno-2-39"></a><span class="c1"># Save and show graph</span>
</span><span id="__span-2-40"><a id="__codelineno-2-40" name="__codelineno-2-40"></a><span class="n">plt</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s2">&quot;combined.HRun.acc.VARIANTS.2L.distribution_DP.png&quot;</span><span class="p">)</span>
</span><span id="__span-2-41"><a id="__codelineno-2-41" name="__codelineno-2-41"></a><span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</span></code></pre></div></td></tr></table></div>
<p>Variants and invariants are filtered using different criteria, reflecting their distinct error profiles and interpretation requirements. For example, thresholds for MQ, FS, and DP are applied only to variants, while a simpler DP filter is applied to invariant sites. The script below performs these filtering steps using GATK’s VariantFiltration after indexing the input VCFs. The DP&lt;300 cutoff, among others, was chosen based on the DP distribution plots generated with the Python script:</p>
<div class="language-bash highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-3-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-3-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-3-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-3-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-3-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-3-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-3-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-3-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-3-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-3-10">10</a></span>
<span class="normal"><a href="#__codelineno-3-11">11</a></span>
<span class="normal"><a href="#__codelineno-3-12">12</a></span>
<span class="normal"><a href="#__codelineno-3-13">13</a></span>
<span class="normal"><a href="#__codelineno-3-14">14</a></span>
<span class="normal"><a href="#__codelineno-3-15">15</a></span>
<span class="normal"><a href="#__codelineno-3-16">16</a></span>
<span class="normal"><a href="#__codelineno-3-17">17</a></span>
<span class="normal"><a href="#__codelineno-3-18">18</a></span>
<span class="normal"><a href="#__codelineno-3-19">19</a></span>
<span class="normal"><a href="#__codelineno-3-20">20</a></span>
<span class="normal"><a href="#__codelineno-3-21">21</a></span>
<span class="normal"><a href="#__codelineno-3-22">22</a></span>
<span class="normal"><a href="#__codelineno-3-23">23</a></span>
<span class="normal"><a href="#__codelineno-3-24">24</a></span>
<span class="normal"><a href="#__codelineno-3-25">25</a></span>
<span class="normal"><a href="#__codelineno-3-26">26</a></span>
<span class="normal"><a href="#__codelineno-3-27">27</a></span>
<span class="normal"><a href="#__codelineno-3-28">28</a></span>
<span class="normal"><a href="#__codelineno-3-29">29</a></span>
<span class="normal"><a href="#__codelineno-3-30">30</a></span>
<span class="normal"><a href="#__codelineno-3-31">31</a></span>
<span class="normal"><a href="#__codelineno-3-32">32</a></span>
<span class="normal"><a href="#__codelineno-3-33">33</a></span>
<span class="normal"><a href="#__codelineno-3-34">34</a></span>
<span class="normal"><a href="#__codelineno-3-35">35</a></span>
<span class="normal"><a href="#__codelineno-3-36">36</a></span>
<span class="normal"><a href="#__codelineno-3-37">37</a></span>
<span class="normal"><a href="#__codelineno-3-38">38</a></span>
<span class="normal"><a href="#__codelineno-3-39">39</a></span>
<span class="normal"><a href="#__codelineno-3-40">40</a></span>
<span class="normal"><a href="#__codelineno-3-41">41</a></span>
<span class="normal"><a href="#__codelineno-3-42">42</a></span>
<span class="normal"><a href="#__codelineno-3-43">43</a></span>
<span class="normal"><a href="#__codelineno-3-44">44</a></span>
<span class="normal"><a href="#__codelineno-3-45">45</a></span>
<span class="normal"><a href="#__codelineno-3-46">46</a></span>
<span class="normal"><a href="#__codelineno-3-47">47</a></span>
<span class="normal"><a href="#__codelineno-3-48">48</a></span>
<span class="normal"><a href="#__codelineno-3-49">49</a></span>
<span class="normal"><a href="#__codelineno-3-50">50</a></span>
<span class="normal"><a href="#__codelineno-3-51">51</a></span>
<span class="normal"><a href="#__codelineno-3-52">52</a></span>
<span class="normal"><a href="#__codelineno-3-53">53</a></span>
<span class="normal"><a href="#__codelineno-3-54">54</a></span>
<span class="normal"><a href="#__codelineno-3-55">55</a></span>
<span class="normal"><a href="#__codelineno-3-56">56</a></span>
<span class="normal"><a href="#__codelineno-3-57">57</a></span>
<span class="normal"><a href="#__codelineno-3-58">58</a></span>
<span class="normal"><a href="#__codelineno-3-59">59</a></span>
<span class="normal"><a href="#__codelineno-3-60">60</a></span>
<span class="normal"><a href="#__codelineno-3-61">61</a></span></pre></div></td><td class="code"><div><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2"></a><span class="c1">###################configuration slurm##############################</span>
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3"></a><span class="c1">#SBATCH -A invalbo</span>
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4"></a><span class="c1">#SBATCH --time=01:00:00</span>
</span><span id="__span-3-5"><a id="__codelineno-3-5" name="__codelineno-3-5"></a><span class="c1">#SBATCH --job-name=vfiltABAG</span>
</span><span id="__span-3-6"><a id="__codelineno-3-6" name="__codelineno-3-6"></a><span class="c1">#SBATCH -p fast</span>
</span><span id="__span-3-7"><a id="__codelineno-3-7" name="__codelineno-3-7"></a><span class="c1">#SBATCH --nodes=1</span>
</span><span id="__span-3-8"><a id="__codelineno-3-8" name="__codelineno-3-8"></a><span class="c1">#SBATCH --ntasks=1</span>
</span><span id="__span-3-9"><a id="__codelineno-3-9" name="__codelineno-3-9"></a><span class="c1">#SBATCH --cpus-per-task=2</span>
</span><span id="__span-3-10"><a id="__codelineno-3-10" name="__codelineno-3-10"></a><span class="c1">#SBATCH --mem=16G</span>
</span><span id="__span-3-11"><a id="__codelineno-3-11" name="__codelineno-3-11"></a><span class="c1">#SBATCH --array=2</span>
</span><span id="__span-3-12"><a id="__codelineno-3-12" name="__codelineno-3-12"></a><span class="c1">#SBATCH --output=Cluster_logs/%x-%j-%N.out</span>
</span><span id="__span-3-13"><a id="__codelineno-3-13" name="__codelineno-3-13"></a><span class="c1">#SBATCH --error=Cluster_logs/%x-%j-%N.err</span>
</span><span id="__span-3-14"><a id="__codelineno-3-14" name="__codelineno-3-14"></a><span class="c1">#SBATCH --mail-user=loic.talignani@ird.fr</span>
</span><span id="__span-3-15"><a id="__codelineno-3-15" name="__codelineno-3-15"></a><span class="c1">#SBATCH --mail-type=FAIL</span>
</span><span id="__span-3-16"><a id="__codelineno-3-16" name="__codelineno-3-16"></a><span class="c1">####################################################################</span>
</span><span id="__span-3-17"><a id="__codelineno-3-17" name="__codelineno-3-17"></a>
</span><span id="__span-3-18"><a id="__codelineno-3-18" name="__codelineno-3-18"></a><span class="c1"># Les VCFs ont été filtrés par accessibilité. Il est donc inutile de filtrer les HighDP.</span>
</span><span id="__span-3-19"><a id="__codelineno-3-19" name="__codelineno-3-19"></a>
</span><span id="__span-3-20"><a id="__codelineno-3-20" name="__codelineno-3-20"></a><span class="nv">start_time</span><span class="o">=</span><span class="k">$(</span>date<span class="w"> </span>+%s<span class="k">)</span>
</span><span id="__span-3-21"><a id="__codelineno-3-21" name="__codelineno-3-21"></a>
</span><span id="__span-3-22"><a id="__codelineno-3-22" name="__codelineno-3-22"></a><span class="c1"># Récupération du nom du fichier BAM à partir du fichier de liste</span>
</span><span id="__span-3-23"><a id="__codelineno-3-23" name="__codelineno-3-23"></a><span class="nv">CHROM</span><span class="o">=</span><span class="k">$(</span>sed<span class="w"> </span>-n<span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span><span class="s2">p&quot;</span><span class="w"> </span>info_files/chrom.list<span class="k">)</span>
</span><span id="__span-3-24"><a id="__codelineno-3-24" name="__codelineno-3-24"></a>
</span><span id="__span-3-25"><a id="__codelineno-3-25" name="__codelineno-3-25"></a><span class="c1"># Répertoire de sortie pour les fichiers</span>
</span><span id="__span-3-26"><a id="__codelineno-3-26" name="__codelineno-3-26"></a><span class="nv">OUTPUT_DIR</span><span class="o">=</span><span class="s2">&quot;vcfs_flt_HRun_slct_vrt&quot;</span>
</span><span id="__span-3-27"><a id="__codelineno-3-27" name="__codelineno-3-27"></a>
</span><span id="__span-3-28"><a id="__codelineno-3-28" name="__codelineno-3-28"></a><span class="nv">LOG_DIR</span><span class="o">=</span><span class="s2">&quot;Cluster_logs&quot;</span>
</span><span id="__span-3-29"><a id="__codelineno-3-29" name="__codelineno-3-29"></a>mkdir<span class="w"> </span>-p<span class="w"> </span><span class="si">${</span><span class="nv">LOG_DIR</span><span class="si">}</span>
</span><span id="__span-3-30"><a id="__codelineno-3-30" name="__codelineno-3-30"></a>
</span><span id="__span-3-31"><a id="__codelineno-3-31" name="__codelineno-3-31"></a><span class="c1"># Load module</span>
</span><span id="__span-3-32"><a id="__codelineno-3-32" name="__codelineno-3-32"></a>module<span class="w"> </span>load<span class="w"> </span>gatk4/4.2.6.1
</span><span id="__span-3-33"><a id="__codelineno-3-33" name="__codelineno-3-33"></a>
</span><span id="__span-3-34"><a id="__codelineno-3-34" name="__codelineno-3-34"></a>gatk<span class="w"> </span>IndexFeatureFile<span class="w"> </span>-I<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.VARIANTS.vcf.gz
</span><span id="__span-3-35"><a id="__codelineno-3-35" name="__codelineno-3-35"></a>gatk<span class="w"> </span>IndexFeatureFile<span class="w"> </span>-I<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.INVARIANTS.vcf.gz
</span><span id="__span-3-36"><a id="__codelineno-3-36" name="__codelineno-3-36"></a>
</span><span id="__span-3-37"><a id="__codelineno-3-37" name="__codelineno-3-37"></a>gatk<span class="w"> </span>VariantFiltration<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-38"><a id="__codelineno-3-38" name="__codelineno-3-38"></a><span class="w"> </span>-R<span class="w"> </span>resources/genome/Anopheles-gambiae-PEST_CHROMOSOMES_AgamP4.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-39"><a id="__codelineno-3-39" name="__codelineno-3-39"></a><span class="w"> </span>-O<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.VARIANTS.flag.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-40"><a id="__codelineno-3-40" name="__codelineno-3-40"></a><span class="w"> </span>-V<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.VARIANTS.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-41"><a id="__codelineno-3-41" name="__codelineno-3-41"></a><span class="w"> </span>--filter-expression<span class="w"> </span><span class="s2">&quot;MQ&lt;40.0&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-42"><a id="__codelineno-3-42" name="__codelineno-3-42"></a><span class="w"> </span>--filter-name<span class="w"> </span><span class="s2">&quot;MQfilter&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-43"><a id="__codelineno-3-43" name="__codelineno-3-43"></a><span class="w"> </span>--filter-expression<span class="w"> </span><span class="s2">&quot;DP&lt;300&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-44"><a id="__codelineno-3-44" name="__codelineno-3-44"></a><span class="w"> </span>--filter-name<span class="w"> </span><span class="s2">&quot;LowDP&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-45"><a id="__codelineno-3-45" name="__codelineno-3-45"></a><span class="w"> </span>--filter-expression<span class="w"> </span><span class="s2">&quot;FS&gt;60.0&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-46"><a id="__codelineno-3-46" name="__codelineno-3-46"></a><span class="w"> </span>--filter-name<span class="w"> </span><span class="s2">&quot;FSfilter&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-47"><a id="__codelineno-3-47" name="__codelineno-3-47"></a><span class="w"> </span>--missing-values-evaluate-as-failing<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-48"><a id="__codelineno-3-48" name="__codelineno-3-48"></a><span class="w"> </span><span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="si">${</span><span class="nv">LOG_DIR</span><span class="si">}</span>/combined.chr<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.acc.HRun.VARIANTS.flag.log
</span><span id="__span-3-49"><a id="__codelineno-3-49" name="__codelineno-3-49"></a>
</span><span id="__span-3-50"><a id="__codelineno-3-50" name="__codelineno-3-50"></a>gatk<span class="w"> </span>VariantFiltration<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-51"><a id="__codelineno-3-51" name="__codelineno-3-51"></a><span class="w"> </span>-R<span class="w"> </span>resources/genome/Anopheles-gambiae-PEST_CHROMOSOMES_AgamP4.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-52"><a id="__codelineno-3-52" name="__codelineno-3-52"></a><span class="w"> </span>-O<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/AB.AG.HC.acc.INVARIANTS.flag.<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-53"><a id="__codelineno-3-53" name="__codelineno-3-53"></a><span class="w"> </span>-V<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/AB.AG.HC.acc.INVARIANTS.<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-54"><a id="__codelineno-3-54" name="__codelineno-3-54"></a><span class="w"> </span>--filter-expression<span class="w"> </span><span class="s2">&quot;DP&lt;300&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-55"><a id="__codelineno-3-55" name="__codelineno-3-55"></a><span class="w"> </span>--filter-name<span class="w"> </span><span class="s2">&quot;LowDP&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-56"><a id="__codelineno-3-56" name="__codelineno-3-56"></a><span class="w"> </span>--missing-values-evaluate-as-failing<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-57"><a id="__codelineno-3-57" name="__codelineno-3-57"></a><span class="w"> </span><span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="si">${</span><span class="nv">LOG_DIR</span><span class="si">}</span>/AB.AG.HC.acc.INVARIANTS.flag.<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.log
</span><span id="__span-3-58"><a id="__codelineno-3-58" name="__codelineno-3-58"></a>
</span><span id="__span-3-59"><a id="__codelineno-3-59" name="__codelineno-3-59"></a><span class="nv">end_time</span><span class="o">=</span><span class="k">$(</span>date<span class="w"> </span>+%s<span class="k">)</span><span class="w">  </span><span class="c1"># Capture l&#39;heure de fin</span>
</span><span id="__span-3-60"><a id="__codelineno-3-60" name="__codelineno-3-60"></a><span class="nv">duration</span><span class="o">=</span><span class="k">$((</span><span class="nv">end_time</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="nv">start_time</span><span class="k">))</span><span class="w">  </span><span class="c1"># Calcule la durée</span>
</span><span id="__span-3-61"><a id="__codelineno-3-61" name="__codelineno-3-61"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;The script completed successfully in </span><span class="k">$((</span><span class="nv">duration</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="m">60</span><span class="k">))</span><span class="s2"> minutes and </span><span class="k">$((</span><span class="nv">duration</span><span class="w"> </span><span class="o">%</span><span class="w"> </span><span class="m">60</span><span class="k">))</span><span class="s2"> seconds.&quot;</span>
</span></code></pre></div></td></tr></table></div>
<h4 id="filtering-criteria-analysis">Filtering Criteria Analysis<a class="headerlink" href="#filtering-criteria-analysis" title="Permanent link">&para;</a></h4>
<p><strong>Variant Site Filters:</strong><br />
- <strong>MQ &lt; 40.0</strong>: Removes sites with poor mapping quality, indicating potential alignment artifacts<br />
- <strong>DP &lt; 300</strong>: Excludes undercovered sites with insufficient statistical power for genotype calling<br />
- <strong>FS &gt; 60.0</strong>: Filters sites with excessive strand bias, suggesting technical artifacts<br />
- <strong>--missing-values-evaluate-as-failing</strong>: Treats missing annotation values as failing filters  </p>
<p><strong>Invariant Site Filters:</strong><br />
- <strong>DP &lt; 300</strong>: Applies only depth filtering, as other quality metrics are less relevant for monomorphic sites<br />
- <strong>Simplified criteria</strong>: Reflects the lower complexity of invariant site assessment  </p>
<h4 id="quality-metric-explanations">Quality Metric Explanations<a class="headerlink" href="#quality-metric-explanations" title="Permanent link">&para;</a></h4>
<p><strong>Mapping Quality (MQ):</strong><br />
- Root mean square of mapping qualities of reads supporting the variant<br />
- Values &lt; 40 indicate poor uniqueness of genomic position<br />
- Critical for excluding paralogous sequence alignments  </p>
<p><strong>Fisher Strand Bias (FS):</strong><br />
- Phred-scaled probability of strand bias in variant calls<br />
- Values &gt; 60 suggest systematic sequencing artifacts<br />
- Particularly important for distinguishing true variants from technical noise  </p>
<p><strong>Depth (DP):</strong><br />
- Total read depth across all samples at the site<br />
- Minimum threshold ensures adequate statistical power<br />
- Balances sensitivity with computational efficiency  </p>
<h3 id="step-4-final-filtering-and-site-selection">Step 4: Final Filtering and Site Selection<a class="headerlink" href="#step-4-final-filtering-and-site-selection" title="Permanent link">&para;</a></h3>
<p>The final step removes all flagged sites, retaining only high-quality variants and invariants:</p>
<div class="language-bash highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-4-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-4-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-4-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-4-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-4-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-4-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-4-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-4-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-4-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-4-10">10</a></span>
<span class="normal"><a href="#__codelineno-4-11">11</a></span>
<span class="normal"><a href="#__codelineno-4-12">12</a></span>
<span class="normal"><a href="#__codelineno-4-13">13</a></span>
<span class="normal"><a href="#__codelineno-4-14">14</a></span>
<span class="normal"><a href="#__codelineno-4-15">15</a></span>
<span class="normal"><a href="#__codelineno-4-16">16</a></span>
<span class="normal"><a href="#__codelineno-4-17">17</a></span>
<span class="normal"><a href="#__codelineno-4-18">18</a></span>
<span class="normal"><a href="#__codelineno-4-19">19</a></span>
<span class="normal"><a href="#__codelineno-4-20">20</a></span>
<span class="normal"><a href="#__codelineno-4-21">21</a></span>
<span class="normal"><a href="#__codelineno-4-22">22</a></span>
<span class="normal"><a href="#__codelineno-4-23">23</a></span>
<span class="normal"><a href="#__codelineno-4-24">24</a></span>
<span class="normal"><a href="#__codelineno-4-25">25</a></span>
<span class="normal"><a href="#__codelineno-4-26">26</a></span>
<span class="normal"><a href="#__codelineno-4-27">27</a></span>
<span class="normal"><a href="#__codelineno-4-28">28</a></span>
<span class="normal"><a href="#__codelineno-4-29">29</a></span>
<span class="normal"><a href="#__codelineno-4-30">30</a></span>
<span class="normal"><a href="#__codelineno-4-31">31</a></span>
<span class="normal"><a href="#__codelineno-4-32">32</a></span>
<span class="normal"><a href="#__codelineno-4-33">33</a></span>
<span class="normal"><a href="#__codelineno-4-34">34</a></span>
<span class="normal"><a href="#__codelineno-4-35">35</a></span>
<span class="normal"><a href="#__codelineno-4-36">36</a></span>
<span class="normal"><a href="#__codelineno-4-37">37</a></span>
<span class="normal"><a href="#__codelineno-4-38">38</a></span>
<span class="normal"><a href="#__codelineno-4-39">39</a></span>
<span class="normal"><a href="#__codelineno-4-40">40</a></span>
<span class="normal"><a href="#__codelineno-4-41">41</a></span>
<span class="normal"><a href="#__codelineno-4-42">42</a></span>
<span class="normal"><a href="#__codelineno-4-43">43</a></span>
<span class="normal"><a href="#__codelineno-4-44">44</a></span>
<span class="normal"><a href="#__codelineno-4-45">45</a></span>
<span class="normal"><a href="#__codelineno-4-46">46</a></span>
<span class="normal"><a href="#__codelineno-4-47">47</a></span>
<span class="normal"><a href="#__codelineno-4-48">48</a></span></pre></div></td><td class="code"><div><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2"></a><span class="c1">###################configuration slurm##############################</span>
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3"></a><span class="c1">#SBATCH -A invalbo</span>
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4"></a><span class="c1">#SBATCH --time=01:00:00</span>
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5"></a><span class="c1">#SBATCH --job-name=snpfltABAG</span>
</span><span id="__span-4-6"><a id="__codelineno-4-6" name="__codelineno-4-6"></a><span class="c1">#SBATCH -p fast</span>
</span><span id="__span-4-7"><a id="__codelineno-4-7" name="__codelineno-4-7"></a><span class="c1">#SBATCH --nodes=1</span>
</span><span id="__span-4-8"><a id="__codelineno-4-8" name="__codelineno-4-8"></a><span class="c1">#SBATCH --ntasks=1</span>
</span><span id="__span-4-9"><a id="__codelineno-4-9" name="__codelineno-4-9"></a><span class="c1">#SBATCH --cpus-per-task=8</span>
</span><span id="__span-4-10"><a id="__codelineno-4-10" name="__codelineno-4-10"></a><span class="c1">#SBATCH --mem=16G</span>
</span><span id="__span-4-11"><a id="__codelineno-4-11" name="__codelineno-4-11"></a><span class="c1">#SBATCH --array=2</span>
</span><span id="__span-4-12"><a id="__codelineno-4-12" name="__codelineno-4-12"></a><span class="c1">#SBATCH --output=Cluster_logs/%x-%j-%N.out</span>
</span><span id="__span-4-13"><a id="__codelineno-4-13" name="__codelineno-4-13"></a><span class="c1">#SBATCH --error=Cluster_logs/%x-%j-%N.err</span>
</span><span id="__span-4-14"><a id="__codelineno-4-14" name="__codelineno-4-14"></a><span class="c1">#SBATCH --mail-user=loic.talignani@ird.fr</span>
</span><span id="__span-4-15"><a id="__codelineno-4-15" name="__codelineno-4-15"></a><span class="c1">#SBATCH --mail-type=FAIL</span>
</span><span id="__span-4-16"><a id="__codelineno-4-16" name="__codelineno-4-16"></a><span class="c1">####################################################################</span>
</span><span id="__span-4-17"><a id="__codelineno-4-17" name="__codelineno-4-17"></a>
</span><span id="__span-4-18"><a id="__codelineno-4-18" name="__codelineno-4-18"></a><span class="nv">start_time</span><span class="o">=</span><span class="k">$(</span>date<span class="w"> </span>+%s<span class="k">)</span>
</span><span id="__span-4-19"><a id="__codelineno-4-19" name="__codelineno-4-19"></a>
</span><span id="__span-4-20"><a id="__codelineno-4-20" name="__codelineno-4-20"></a><span class="c1"># Retrait des SNPs n&#39;étant pas au statut FILTER=PASS</span>
</span><span id="__span-4-21"><a id="__codelineno-4-21" name="__codelineno-4-21"></a>
</span><span id="__span-4-22"><a id="__codelineno-4-22" name="__codelineno-4-22"></a><span class="c1"># Récupération du nom du fichier BAM à partir du fichier de liste</span>
</span><span id="__span-4-23"><a id="__codelineno-4-23" name="__codelineno-4-23"></a><span class="nv">CHROM</span><span class="o">=</span><span class="k">$(</span>sed<span class="w"> </span>-n<span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span><span class="s2">p&quot;</span><span class="w"> </span>info_files/chroms_list<span class="k">)</span>
</span><span id="__span-4-24"><a id="__codelineno-4-24" name="__codelineno-4-24"></a>
</span><span id="__span-4-25"><a id="__codelineno-4-25" name="__codelineno-4-25"></a><span class="c1"># Répertoire de sortie pour les fichiers</span>
</span><span id="__span-4-26"><a id="__codelineno-4-26" name="__codelineno-4-26"></a><span class="nv">OUTPUT_DIR</span><span class="o">=</span>results/05_Variants
</span><span id="__span-4-27"><a id="__codelineno-4-27" name="__codelineno-4-27"></a>
</span><span id="__span-4-28"><a id="__codelineno-4-28" name="__codelineno-4-28"></a><span class="nv">LOG_DIR</span><span class="o">=</span><span class="s2">&quot;results/11_Reports/snpfiltering&quot;</span>
</span><span id="__span-4-29"><a id="__codelineno-4-29" name="__codelineno-4-29"></a>mkdir<span class="w"> </span>-p<span class="w"> </span><span class="si">${</span><span class="nv">LOG_DIR</span><span class="si">}</span>
</span><span id="__span-4-30"><a id="__codelineno-4-30" name="__codelineno-4-30"></a>
</span><span id="__span-4-31"><a id="__codelineno-4-31" name="__codelineno-4-31"></a><span class="c1"># Load module</span>
</span><span id="__span-4-32"><a id="__codelineno-4-32" name="__codelineno-4-32"></a>module<span class="w"> </span>load<span class="w"> </span>gatk4/4.2.6.1
</span><span id="__span-4-33"><a id="__codelineno-4-33" name="__codelineno-4-33"></a>
</span><span id="__span-4-34"><a id="__codelineno-4-34" name="__codelineno-4-34"></a>gatk<span class="w"> </span>SelectVariants<span class="w"> </span><span class="se">\</span>
</span><span id="__span-4-35"><a id="__codelineno-4-35" name="__codelineno-4-35"></a><span class="w"> </span>-V<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/AB.AG.HC.acc.VARIANTS.flag.<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-4-36"><a id="__codelineno-4-36" name="__codelineno-4-36"></a><span class="w"> </span>--exclude-filtered<span class="w"> </span><span class="se">\</span>
</span><span id="__span-4-37"><a id="__codelineno-4-37" name="__codelineno-4-37"></a><span class="w"> </span>-O<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/AB.AG.HC.acc.VARIANTS.flag.pass.<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-4-38"><a id="__codelineno-4-38" name="__codelineno-4-38"></a><span class="w"> </span><span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="si">${</span><span class="nv">LOG_DIR</span><span class="si">}</span>/AB.AG.HC.acc.VARIANTS.flag.pass.<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.log
</span><span id="__span-4-39"><a id="__codelineno-4-39" name="__codelineno-4-39"></a>
</span><span id="__span-4-40"><a id="__codelineno-4-40" name="__codelineno-4-40"></a>gatk<span class="w"> </span>SelectVariants<span class="w"> </span><span class="se">\</span>
</span><span id="__span-4-41"><a id="__codelineno-4-41" name="__codelineno-4-41"></a><span class="w"> </span>-V<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>//AB.AG.HC.acc.INVARIANTS.flag.<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-4-42"><a id="__codelineno-4-42" name="__codelineno-4-42"></a><span class="w"> </span>--exclude-filtered<span class="w"> </span><span class="se">\</span>
</span><span id="__span-4-43"><a id="__codelineno-4-43" name="__codelineno-4-43"></a><span class="w"> </span>-O<span class="w"> </span><span class="si">${</span><span class="nv">OUTPUT_DIR</span><span class="si">}</span>/AB.AG.HC.acc.INVARIANTS.flag.pass.<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-4-44"><a id="__codelineno-4-44" name="__codelineno-4-44"></a><span class="w"> </span><span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="si">${</span><span class="nv">LOG_DIR</span><span class="si">}</span>/AB.AG.HC.acc.INVARIANTS.flag.pass.<span class="si">${</span><span class="nv">CHROM</span><span class="si">}</span>.log
</span><span id="__span-4-45"><a id="__codelineno-4-45" name="__codelineno-4-45"></a>
</span><span id="__span-4-46"><a id="__codelineno-4-46" name="__codelineno-4-46"></a><span class="nv">end_time</span><span class="o">=</span><span class="k">$(</span>date<span class="w"> </span>+%s<span class="k">)</span><span class="w">  </span><span class="c1"># Capture l&#39;heure de fin</span>
</span><span id="__span-4-47"><a id="__codelineno-4-47" name="__codelineno-4-47"></a><span class="nv">duration</span><span class="o">=</span><span class="k">$((</span><span class="nv">end_time</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="nv">start_time</span><span class="k">))</span><span class="w">  </span><span class="c1"># Calcule la durée</span>
</span><span id="__span-4-48"><a id="__codelineno-4-48" name="__codelineno-4-48"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;The script completed successfully in </span><span class="k">$((</span><span class="nv">duration</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="m">60</span><span class="k">))</span><span class="s2"> minutes and </span><span class="k">$((</span><span class="nv">duration</span><span class="w"> </span><span class="o">%</span><span class="w"> </span><span class="m">60</span><span class="k">))</span><span class="s2"> seconds.&quot;</span>
</span></code></pre></div></td></tr></table></div>
<h4 id="final-selection-parameters">Final Selection Parameters<a class="headerlink" href="#final-selection-parameters" title="Permanent link">&para;</a></h4>
<p><strong>--exclude-filtered</strong>: Removes all sites that failed any filter criterion, retaining only sites marked as "PASS"  </p>
<p><strong>Quality Control Verification</strong>: Final datasets should be assessed for:<br />
- Retained site counts and genomic distribution<br />
- Quality metric distributions post-filtering<br />
- Transition/transversion ratios (variants)<br />
- Coverage uniformity across chromosomes  </p>
<h2 id="population-genomics-applications">Population Genomics Applications<a class="headerlink" href="#population-genomics-applications" title="Permanent link">&para;</a></h2>
<h3 id="downstream-analysis-compatibility">Downstream Analysis Compatibility<a class="headerlink" href="#downstream-analysis-compatibility" title="Permanent link">&para;</a></h3>
<p><strong>Nucleotide Diversity Calculations:</strong>
<div class="language-bash highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-5-1">1</a></span>
<span class="normal"><a href="#__codelineno-5-2">2</a></span>
<span class="normal"><a href="#__codelineno-5-3">3</a></span>
<span class="normal"><a href="#__codelineno-5-4">4</a></span></pre></div></td><td class="code"><div><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1"></a><span class="c1"># Example calculation using both datasets</span>
</span><span id="__span-5-2"><a id="__codelineno-5-2" name="__codelineno-5-2"></a><span class="nv">total_sites</span><span class="o">=</span><span class="k">$(</span>bcftools<span class="w"> </span>view<span class="w"> </span>-H<span class="w"> </span>variants.pass.vcf.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>wc<span class="w"> </span>-l<span class="k">)</span><span class="w"> </span>+<span class="w"> </span><span class="k">$(</span>bcftools<span class="w"> </span>view<span class="w"> </span>-H<span class="w"> </span>invariants.pass.vcf.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>wc<span class="w"> </span>-l<span class="k">)</span>
</span><span id="__span-5-3"><a id="__codelineno-5-3" name="__codelineno-5-3"></a><span class="nv">polymorphic_sites</span><span class="o">=</span><span class="k">$(</span>bcftools<span class="w"> </span>view<span class="w"> </span>-H<span class="w"> </span>variants.pass.vcf.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>wc<span class="w"> </span>-l<span class="k">)</span>
</span><span id="__span-5-4"><a id="__codelineno-5-4" name="__codelineno-5-4"></a><span class="nv">pi</span><span class="o">=</span><span class="k">$(</span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;scale=6; </span><span class="nv">$polymorphic_sites</span><span class="s2"> / </span><span class="nv">$total_sites</span><span class="s2">&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>bc<span class="k">)</span>
</span></code></pre></div></td></tr></table></div></p>
<p><strong>Site Frequency Spectrum Analysis:</strong><br />
- Variants provide polymorphic site frequencies<br />
- Invariants contribute to the monomorphic class (frequency = 0 or 1)<br />
- Combined datasets enable accurate demographic inference  </p>
<p><strong>Selective Sweep Detection:</strong><br />
- Reduced diversity in invariant sites indicates purifying selection<br />
- Elevated diversity in variant sites suggests balancing selection<br />
- Comparative analysis identifies selection signatures  </p>
<h3 id="computational-considerations">Computational Considerations<a class="headerlink" href="#computational-considerations" title="Permanent link">&para;</a></h3>
<p><strong>Storage Requirements:</strong><br />
- Invariant sites typically outnumber variants by 10-100 fold<br />
- Compressed VCF formats essential for storage efficiency<br />
- Indexed files enable rapid region-specific access  </p>
<p><strong>Processing Efficiency:</strong><br />
- Parallel chromosome processing reduces computational time<br />
- Separate filtering pipelines optimize resource utilization<br />
- Modular workflow facilitates parameter optimization  </p>
<h2 id="best-practices-and-recommendations">Best Practices and Recommendations<a class="headerlink" href="#best-practices-and-recommendations" title="Permanent link">&para;</a></h2>
<h3 id="filter-threshold-selection">Filter Threshold Selection<a class="headerlink" href="#filter-threshold-selection" title="Permanent link">&para;</a></h3>
<p><strong>Conservative Approach:</strong><br />
- Higher depth requirements (DP &gt; 500) for critical analyses<br />
- Stricter quality thresholds (MQ &gt; 50) for publication-quality datasets<br />
- Additional filters (BaseQRankSum, ReadPosRankSum) for variant sites  </p>
<p><strong>Moderate Approach:</strong><br />
- Balanced sensitivity and specificity (current implementation)<br />
- Suitable for most population genetic analyses <br />
- Adequate for demographic inference and diversity estimation  </p>
<p><strong>Lenient Approach:</strong><br />
- Lower thresholds (DP &gt; 100) for sample-limited studies<br />
- Relaxed quality criteria for exploratory analyses  <br />
- Requires additional downstream validation  </p>
<h3 id="quality-control-validation">Quality Control Validation<a class="headerlink" href="#quality-control-validation" title="Permanent link">&para;</a></h3>
<p><strong>Statistical Assessment:</strong><br />
1. <strong>Ti/Tv ratios</strong>: Expected values ~2.0-2.1 for human, variable for other species<br />
2. <strong>Depth distributions</strong>: Should follow expected coverage patterns<br />
3. <strong>Allele frequency spectra</strong>: Should match population genetic expectations<br />
4. <strong>Genomic distribution</strong>: Uniform distribution across accessible regions  </p>
<p><strong>Comparative Analysis:</strong><br />
- Benchmark against high-quality reference datasets<br />
- Cross-platform validation using orthogonal technologies<br />
- Population-specific optimization based on demographic history  </p>
<h3 id="integration-strategies">Integration Strategies<a class="headerlink" href="#integration-strategies" title="Permanent link">&para;</a></h3>
<p><strong>Downstream Merging:</strong>
<div class="language-bash highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-6-1">1</a></span>
<span class="normal"><a href="#__codelineno-6-2">2</a></span>
<span class="normal"><a href="#__codelineno-6-3">3</a></span>
<span class="normal"><a href="#__codelineno-6-4">4</a></span>
<span class="normal"><a href="#__codelineno-6-5">5</a></span></pre></div></td><td class="code"><div><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1"></a><span class="c1"># Combine filtered variants and invariants for analysis</span>
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2"></a>bcftools<span class="w"> </span>concat<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3"></a><span class="w">  </span>variants.pass.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4"></a><span class="w">  </span>invariants.pass.vcf.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-5"><a id="__codelineno-6-5" name="__codelineno-6-5"></a><span class="w">  </span>-a<span class="w"> </span>-D<span class="w"> </span>-Oz<span class="w"> </span>-o<span class="w"> </span>combined.filtered.vcf.gz
</span></code></pre></div></td></tr></table></div></p>
<p><strong>Analysis-Specific Subsets:</strong><br />
- Coding regions: Enhanced filtering for functional impact assessment<br />
- Intergenic regions: Relaxed filtering for neutral evolution studies<br />
- Regulatory elements: Intermediate filtering for selection analysis  </p>
<h2 id="limitations-and-considerations">Limitations and Considerations<a class="headerlink" href="#limitations-and-considerations" title="Permanent link">&para;</a></h2>
<h3 id="technical-limitations">Technical Limitations<a class="headerlink" href="#technical-limitations" title="Permanent link">&para;</a></h3>
<p><strong>Ascertainment Bias:</strong><br />
- Variant discovery influenced by reference genome quality<br />
- Population-specific variants may be systematically missed<br />
- Affects demographic inference accuracy  </p>
<p><strong>Coverage Heterogeneity:</strong><br />
- Uneven sequencing depth across genomic regions<br />
- Systematic bias in variant detection sensitivity<br />
- Requires careful normalization in comparative studies  </p>
<h3 id="biological-considerations">Biological Considerations<a class="headerlink" href="#biological-considerations" title="Permanent link">&para;</a></h3>
<p><strong>Population Structure:</strong><br />
- Filtering thresholds may need population-specific adjustment<br />
- Admixed populations require modified quality criteria<br />
- Demographic history affects optimal parameter selection  </p>
<p><strong>Evolutionary Constraints:</strong><br />
- Functional regions may require different filtering approaches<br />
- Selective pressure influences variant quality distributions<br />
- Conservation levels affect optimal sensitivity/specificity balance  </p>
<h2 id="future-developments">Future Developments<a class="headerlink" href="#future-developments" title="Permanent link">&para;</a></h2>
<h3 id="methodological-improvements">Methodological Improvements<a class="headerlink" href="#methodological-improvements" title="Permanent link">&para;</a></h3>
<p><strong>Machine Learning Approaches:</strong><br />
- Automated filter threshold optimization<br />
- Context-specific quality assessment<br />
- Population-adapted filtering strategies  </p>
<p><strong>Long-Read Integration:</strong><br />
- Improved structural variant detection<br />
- Enhanced repetitive region characterization<br />
- Reduced reference bias in variant calling  </p>
<h3 id="analytical-advances">Analytical Advances<a class="headerlink" href="#analytical-advances" title="Permanent link">&para;</a></h3>
<p><strong>Comprehensive Genomic Datasets:</strong><br />
- Integration of regulatory annotations<br />
- Functional impact prediction<br />
- Evolutionary constraint incorporation  </p>
<p><strong>Population-Specific Resources:</strong><br />
- Ethnically diverse reference panels<br />
- Population-specific quality metrics<br />
- Adaptive filtering frameworks  </p>
<h2 id="conclusion">Conclusion<a class="headerlink" href="#conclusion" title="Permanent link">&para;</a></h2>
<p>The separation and differential filtering of variants and invariants represents a sophisticated approach to population genomic analysis that maximizes both data quality and analytical power. By recognizing the distinct characteristics and requirements of polymorphic and monomorphic sites, this workflow enables accurate estimation of population genetic parameters while maintaining computational efficiency.</p>
<p>The implementation described here provides a robust framework for population genomic studies, with modular components that can be adapted to specific research questions and sample characteristics. The careful balance between sensitivity and specificity, combined with comprehensive quality control measures, ensures the reliability of downstream analyses while preserving the statistical power necessary for demographic inference and selection analysis.</p>
<p>Success in population genomics depends critically on the quality of the underlying data, and the differential treatment of variants and invariants represents a key advancement in achieving this goal. As sequencing technologies and analytical methods continue to evolve, the principles outlined in this workflow will remain essential for extracting meaningful biological insights from genomic variation data.</p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../homopolymer/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Filtering by Homopolymer">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Filtering by Homopolymer
              </div>
            </div>
          </a>
        
        
          
          <a href="../genotype-quality/" class="md-footer__link md-footer__link--next" aria-label="Next: Filtering by Genotype Quality">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Filtering by Genotype Quality
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2025 Loïc Talignani - <a href="#__consent">Change cookie settings</a>

    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/ltalignani/" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://www.linkedin.com/in/loic-talignani-83a7581a2/" target="_blank" rel="noopener" title="www.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3M135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5m282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
      <div class="md-progress" data-md-component="progress" role="progressbar"></div>
    
    
      <div class="md-consent" data-md-component="consent" id="__consent" hidden>
        <div class="md-consent__overlay"></div>
        <aside class="md-consent__inner">
          <form class="md-consent__form md-grid md-typeset" name="consent">
            



<h4>Cookie consent</h4>
<p>We use cookies to recognize your repeated visits and preferences, as well as to measure the effectiveness of our documentation and whether users find what they're searching for. With your consent, you're helping us to make our documentation better.</p>
<input class="md-toggle" type="checkbox" id="__settings" >
<div class="md-consent__settings">
  <ul class="task-list">
    
    
    
      
        
  
  
    
    
  
  <li class="task-list-item">
    <label class="task-list-control">
      <input type="checkbox" name="github" checked>
      <span class="task-list-indicator"></span>
      GitHub
    </label>
  </li>

      
    
    
      
        
  
  
    
    
  
  <li class="task-list-item">
    <label class="task-list-control">
      <input type="checkbox" name="analytics" >
      <span class="task-list-indicator"></span>
      Google Analytics
    </label>
  </li>

      
    
  </ul>
</div>
<div class="md-consent__controls">
  
    
      <button class="md-button md-button--primary">Accept</button>
    
    
    
  
    
    
    
      <label class="md-button" for="__settings">Manage settings</label>
    
  
    
    
      <button type="reset" class="md-button md-button--primary">Reject</button>
    
    
  
</div>
          </form>
        </aside>
      </div>
      <script>var consent=__md_get("__consent");if(consent)for(var input of document.forms.consent.elements)input.name&&(input.checked=consent[input.name]||!1);else"file:"!==location.protocol&&setTimeout((function(){document.querySelector("[data-md-component=consent]").hidden=!1}),250);var form=document.forms.consent;for(var action of["submit","reset"])form.addEventListener(action,(function(e){if(e.preventDefault(),"reset"===e.type)for(var n of document.forms.consent.elements)n.name&&(n.checked=!1);__md_set("__consent",Object.fromEntries(Array.from(new FormData(form).keys()).map((function(e){return[e,!0]})))),location.hash="",location.reload()}))</script>
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.indexes", "navigation.expand", "navigation.path", "navigation.top", "navigation.tabs", "navigation.tabs.sticky", "navigation.sections", "navigation.tracking", "navigation.footer", "navigation.instant", "navigation.instant.progress", "content.code.copy", "content.code.select", "toc.integrate", "toc.follow", "search.suggest", "search.highlight", "search.share"], "search": "../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.56ea9cef.min.js"></script>
      
    
  </body>
</html>