# Welcome to Bash Script Collection


This guide introduces:  

## Basic Bash Commands:
•	[Basic Bash Commands](Basic/basic-bash-commands.md)  
•	[File Management](Basic/file-management.md)  
•	[Text Processing](Basic/text-processing.md)  
•	[Networking](Basic/networking.md)  
•	[Process Management](Basic/process-management.md)  

## Data Organisation
•	[Organising your data](organize-data.md)  
•	[Start Using Git](Git.md)  

## Fastq Files Treatment
•	[The Fastq Format](Fastq/Fastq.md)  
•	[Renaming FASTQ Files Downloaded from the Short Read Archive](Fastq/rename-fastq.md)  
•	[Fastq Quality Control with FastQC](Fastq/FastQC.md)  
•	[Fastq Contamination Checking with Fastq-screen](Fastq/Fastq-screen.md)  
•	[Reads Trimming with Trimmomatic](Fastq/Trimming.md)   
•	[Reads Mapping with BWA](Fastq/BWA.md)  

## SAM/BAM Files 
•	[The SAM/BAM Format](Bam/Bam.md)  
•	[Sorting a SAM/BAM File](Bam/sort-bam.md)  
•	[How to Add or Replace ReadGroup](Bam/add-or-replace-read-groups.md)  
•	[Deduplication](Bam/mark-duplicates.md)  
•	[Merge and deduplicate Bam Files](Bam/merge-and-mark-duplicates.md)  
•	[Indel Realignment](Bam/indel-realignment.md)  
•	[Polishing Bam Files](Bam/polishing.md)    
•	[Bam Quality Control](Bam/bam-qc.md)  
•	[Variant-Calling](Bam/variant-calling.md)  
•	[GenomicsDBImport and GenotypeGVCFs in GATK4 Workflow](Bam/genomicsdbimport.md)  

## VCF Files
•	[The VCF Format](Vcf/Vcf.md)  
•	[Filtering a VCF File by Accessibility](Vcf/accessibility.md)  
•	[Filtering a VCF File by Homopolymer Runs](Vcf/homopolymer.md)  
•	[Filtering Variants & Invariants Sites](Vcf/variants-invariants.md)  
•	[Filtering by Genotype Quality](Vcf/genotype-quality.md)  
•	[VCF concatenation](Vcf/concatenate.md)  
•	[Filtering by GT and Individuals](Vcf/gt-indv.md)  